tests/xfstests/run.pm

```diff
@@ -0,0 +1,176 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: Run tests
+# Maintainer: Nathan Zhao <jtzhao@suse.com>
+package run;
+
+use 5.018;
+use strict;
+use warnings;
+use base "opensusebasetest";
+use File::Basename;
+use testapi;
+use utils;
+use kdump_utils;
+
+# Heartbeat variables
+my $HB_INTVL   = get_var("XFSTESTS_HEARTBEAT_INTERVAL") || 5;
+my $HB_TIMEOUT = get_var("XFSTESTS_HEARTBEAT_TIMEOUT")  || 300;
+my $HB_PATN    = "<heartbeat>";
+my $HB_DONE    = "<done>";
+my $HB_RAMFS   = "/mnt/ramfs";
+my $HB_DONE_FILE = "$HB_RAMFS/test.done";
+my $HB_EXIT_FILE = "$HB_RAMFS/test.exit";
+
+# xfstests variables
+my $WRAPPER  = "/usr/share/qa/qa_test_xfstests/wrapper.sh";
+my $LOG_FILE = "/tmp/xfstests.log";
+
+# blacklist
+my %BLACKLIST = map { $_ => 1 } split(/,/, get_var("XFSTESTS_BLACKLIST"));
+
+
+sub heartbeat_start {
+    # Create ramfs
+    assert_script_run("mkdir -p $HB_RAMFS; umount -f $HB_RAMFS; mount -t ramfs none $HB_RAMFS");
+
+    my $redir = " >> /dev/$serialdev";
+    my $cmd   = "while [[ ! -f $HB_EXIT_FILE ]]; do ";
+    $cmd .= "sleep $HB_INTVL; ";
+    $cmd .= "[[ -f $HB_DONE_FILE ]] && ";
+    $cmd .= "echo \"$HB_DONE\" $redir || echo \"$HB_PATN\" $redir";
+    $cmd .= "; done";
+    script_run("rm -f $HB_DONE_FILE $HB_EXIT_FILE");
+    type_string("nice -n -20 bash -c '$cmd' &\n");
```

Comment:
why you executing this command with type_string ? if it will fail you will not get any feedback about a failure

Reply:
I am curious to see failed test case exactly without this timeout

Reply:
This commands starts a background bash process which will keep running until the end of the test. There's no need to check for its return status.